sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/PointWriterSubpartitionITest.scala

```diff
@@ -0,0 +1,1382 @@
+    }
+
+    // Patch operation will succeed as it only apply increment on numeric type
+    val incrementPartialUpdateValidSchema = StructType(Seq(
+      StructField("propInt", IntegerType),
+      StructField("propLong", LongType),
+      StructField("propFloat", FloatType),
+      StructField("propDouble", DoubleType)
+    ))
+    val patchPartialUpdateItem =
+      CosmosPatchTestHelper.getPatchItemWithSchema(
+        strippedPartitionKeyPath,
+        incrementPartialUpdateValidSchema,
+        originalItem)
+
+    val columnConfigsMap = new TrieMap[String, CosmosPatchColumnConfig]
+    patchPartialUpdateItem.fields().asScala.foreach(field => {
+      columnConfigsMap += field.getKey -> CosmosPatchColumnConfig(
+        field.getKey, CosmosPatchOperationTypes.Increment, s"/${field.getKey}", false)
+    })
+
+    val pointWriterForPatch = CosmosPatchTestHelper.getPointWriterForPatch(columnConfigsMap, container, partitionKeyDefinition)
+
+    pointWriterForPatch.scheduleWrite(partitionKey, patchPartialUpdateItem)
+    pointWriterForPatch.flushAndClose()
+  }
+
+  "Point Writer" should "skip partial update for cosmos system properties" in {
+    val container = getContainer
+    val containerProperties = container.read().block().getProperties
+    val partitionKeyDefinition = containerProperties.getPartitionKeyDefinition
+    val strippedPartitionKeyPath = CosmosPatchTestHelper.getStrippedPartitionKeyPath(partitionKeyDefinition)
+    val writeConfig = CosmosWriteConfig(
+      ItemWriteStrategy.ItemOverwrite,
+      5,
+      bulkEnabled = false)
+
+    val pointWriter = new PointWriter(
+      container, partitionKeyDefinition, writeConfig, DiagnosticsConfig(Option.empty, false, None), MockTaskContext.mockTaskContext())
+
+    // First create one item, as patch can only operate on existing items
+    val itemWithFullSchema = CosmosPatchTestHelper.getPatchItemWithFullSchemaSubpartitions(UUID.randomUUID().toString, strippedPartitionKeyPath)
+    val id = itemWithFullSchema.get("id").textValue()
+    val partitionKey = new PartitionKeyBuilder()
+          .add(itemWithFullSchema.get("tenantId").textValue())
+          .add(itemWithFullSchema.get("userId").textValue())
+          .add(itemWithFullSchema.get("sessionId").textValue())
+          .build()
+
+    pointWriter.scheduleWrite(partitionKey, itemWithFullSchema)
+    pointWriter.flushAndClose()
+    // make sure the item exists
+    val originalItem = container.readItem(id, partitionKey, classOf[ObjectNode]).block().getItem()
+
+    // Cosmos patch does not support for system properties
+    // if we send request to patch for them, server is going to return exception with "Invalid patch request: Cannot patch system property"
+    // so the test is to make sure we have skipped these properties and the request can succeed for other properties
+    val partialUpdateSchema = StructType(Seq(
```

Comment:
Can we also add a validation for the partitionKeyPath -> the path property should be ignored as well

Reply:
I've changed to ignore path for the subpartition tests.